### PR TITLE
[feature-wip](CN Node)Support compute node

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

Issue Number: https://github.com/apache/doris/issues/13144

## Problem summary

Introduce the node role to doris, and the table creation and tablet scheduler will control the storage only assign to the BE nodes.
## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -872,6 +872,8 @@ CONF_mBool(enable_new_load_scan_node, "false");
 // Temp config. True to use new file scanner. Will remove after fully test.
 CONF_mBool(enable_new_file_scanner, "false");
 
+CONF_mString(be_node_type, "be");
```
Comment (morningman) [inline be/src/common/config.h:875]: how about `mixed`, `compute`, etc.
And I think this is not a modifiable config, so it should be `CONF_String`

File: fe/fe-core/src/main/java/org/apache/doris/resource/Tag.java [hunk 2]
```diff
@@ -65,6 +66,7 @@ public class Tag implements Writable {
     public static final String VALUE_DEFAULT_CLUSTER = "default_cluster";
     public static final String VALUE_DEFAULT_TAG = "default";
     public static final String VALUE_INVALID_TAG = "invalid";
+    public static final String VALUE_NODE_TYPE_TAG = "be";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/resource/Tag.java:69]: Actually, there are some predefined field like `TYPE_FUNCTION` and `VALUE_COMPUTATION`, they are supposed to do this feature.
So I think we can use them

File: fe/fe-core/src/main/java/org/apache/doris/common/Config.java [hunk 3]
```diff
@@ -1788,4 +1788,6 @@ public class Config extends ConfigBase {
     @ConfField(mutable = false)
     public static int statistic_task_scheduler_execution_interval_ms = 60 * 1000;
 
+    @ConfField(mutable = true, masterOnly = false)
+    public static int backend_num_for_federation = 3;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/Config.java:1792]: Add some comment for this config

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.76 seconds
 load time: 585 seconds
 storage size: 17154768597 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221019214037_clickbench_pr_31440.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.1 seconds
 load time: 643 seconds
 storage size: 17154810702 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221021044937_clickbench_pr_32318.html

File: docs/zh-CN/docs/sql-manual/sql-reference/Show-Statements/SHOW-BACKENDS.md [hunk 4]
```diff
@@ -52,6 +52,7 @@ SHOW BACKENDS
        10. UsedPct 表示磁盘已使用量百分比。
        11. ErrMsg 用于显示心跳失败时的错误信息。
        12. Status 用于以 JSON 格式显示BE的一些状态信息, 目前包括最后一次BE汇报其tablet的时间信息。
+       13. NodeRole用于展示节点角色, 现在有两种类型: Mix代表原来的节点类型, computation代表只做计算的节点类型.
```
Comment (morningman) [inline docs/zh-CN/docs/sql-manual/sql-reference/Show-Statements/SHOW-BACKENDS.md:55]: I add a new column `HeartbeatFailureCounter` in #13568.
But I forgot to modify the document here.
Could you please add it for me? The content is:

```
13. HeartbeatFailureCounter：现在当前连续失败的心跳次数，如果次数超过 `max_backend_heartbeat_failure_tolerance_count` 配置，则 isAlive 字段会置为 false。
```
English:
```
13. HeartbeatFailureCounter: The current number of heartbeats that have failed consecutively. If the number exceeds the `max_backend_heartbeat_failure_tolerance_count` configuration, the isAlive will be set to false.
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**